prdoc/pr_7723.prdoc

```diff
@@ -0,0 +1,22 @@
+title: '[pallet-bounties] Make BountyUpdatePeriod optional'
+doc:
+- audience: Runtime Dev
+  description: |
+    Refactored `BountyUpdatePeriod` to be an optional configuration parameter.
+    If set to `None`, bounties remain active indefinitely, removing the need for
+    `extend_bounty_expiry` and preventing automatic curator slashing for inactivity.
+
+migrations:
```

Comment:
```suggestion
```
Migration in this sense means storage migration. But there is no storage being migrated.